src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -169,6 +174,10 @@ public void start(final Map<String, String> props) {
                     sourceOffset.put("copy", "true");
                 }
 
+                if (changeStreamDocument.getString("operationType", new BsonString("")).getValue().equalsIgnoreCase(INVALIDATE)) {
+                    cachedResumeToken = changeStreamDocument.getDocument("_id");
```

Comment:
If the cursor was explicitly invalidated then save the resumeToken, so can start after that point.